test/python/http_recorder.py

```diff
@@ -0,0 +1,280 @@
+        if isinstance(data_dict, dict):
+            if map_list[0] in data_dict.keys():
+                if len(map_list) == 1:
+                    return [data_dict]
+                else:
+                    ret.extend(
+                        IdRemoverPersister.get_maching_dicts(data_dict[map_list[0]], map_list[1:]))
+        return ret
+
+    @staticmethod
+    def remove_id_in_a_json(jsonobj, field, path, id_tracker):
+        """
+        Replaces in jsonobj (in-place) the field with dummy value (which is constructed with
+        id_tracker, if it was already reaplced, or path, if it needs to be created).
+
+        Args:
+            jsonobj (dict): json dictionary from the response body
+            field (str): string with the field in the response to by replaced
+            path (str): request path
+            id_tracker (dict): a dictionary of the ids already assigned.
+        """
+
+        map_list = field.split('.')
+        for maching_dict in IdRemoverPersister.get_maching_dicts(jsonobj, map_list):
+            with suppress(KeyError):
+                old_id = maching_dict[map_list[-1]]
+                if old_id not in id_tracker:
+                    new_id = IdRemoverPersister.get_new_id(field, path, id_tracker, type(old_id))
+                    id_tracker[old_id] = new_id
+                maching_dict[map_list[-1]] = id_tracker[old_id]
+
+    @staticmethod
+    def remove_ids_in_a_response(response, fields, path, id_tracker):
+        """
+        Replaces in response (in-place) the fields with dummy values (which is constructed with
+        id_tracker, if it was already reaplced, or path, if it needs to be created).
+
+        Args:
+            response (dict): dictionary of the response body
+            fields (list): list of fields in the response to by replaced
+            path (str): request path
+            id_tracker (dict): a dictionary of the ids already assigned.
+        """
+        body = json.loads(response['body']['string'].decode('utf-8'))
+        for field in fields:
+            IdRemoverPersister.remove_id_in_a_json(body, field, path, id_tracker)
+        response['body']['string'] = json.dumps(body).encode('utf-8')
+
+    @staticmethod
+    def remove_ids(ids2remove, cassette_dict):
+        """
+        Replaces in cassette_dict (in-place) the fields defined by ids2remove with dummy values.
+
+        Args:
+            ids2remove (dict): {request_path: [json_fields]}
+            cassette_dict (dict): a VCR cassette dictionary.
+        """
+
+        id_tracker = {}  # {old_id: new_id}
+        for path, fields in ids2remove.items():
+            responses = IdRemoverPersister.get_responses_with(path, cassette_dict)
+            for response in responses:
+                IdRemoverPersister.remove_ids_in_a_response(response, fields, path, id_tracker)
+        for old_id, new_id in id_tracker.items():
+            if not isinstance(old_id, str):
```

Comment:
Non-blocking: making this test positive and putting the remaining block inside make it clear you want to do replace only textual values.